engine/src/main/java/org/hibernate/validator/internal/engine/valueextraction/ValueExtractorManager.java

```diff
@@ -37,6 +37,7 @@
 /**
  * @author Gunnar Morling
  * @author Guillaume Smet
+ * @author Marko Bekhta
  */
 public class ValueExtractorManager {
```

Comment:
So... about this one... I'm wondering if we should have the `ValueExtractorManager` and have a separate class named `ValueExtractorResolver` that would contain all the resolving logic (and the cache directly in it i.e. `ValueExtractorResolver` would replace `ValueExtractorResolutionCache`). We would keep the `ValueExtractorManager` as the entry point that delegates the calls to the resolver (I don't want us to carry 2 objects everywhere).

I'm wondering if we could make the methods used for container element constraints also use the cache, using the same methods, just having declaredType == runtimeType? Not sure it would work but I prefer to ask. It would be nicer I think (I would keep separate shortcut public methods though).

I think it would also be nice for each public methods involved in the resolution to have comments explaining when it is used (container element constraints/cascaded validation on container element/cascaded validation on the whole container).